File: sdk/elixir/dagger_codegen/lib/dagger/codegen/introspection/types/enum_value.ex
defmodule Dagger.Codegen.Introspection.Types.EnumValue do
  defstruct [
    :deprecation_reason,
    :description,
    :is_deprecated,
    :name,
    :directives
  ]

  def from_map(
        %{
          "deprecationReason" => deprecation_reason,
          "description" => description,
          "isDeprecated" => is_deprecated,
          "name" => name
        } = enum_value
      ) do
    %__MODULE__{
      deprecation_reason: deprecation_reason,
      description: description,
      is_deprecated: is_deprecated,
      name: name,
      directives:
        Enum.map(
          enum_value["directives"] || [],
          &Dagger.Codegen.Introspection.Types.Directive.from_map/1
        )
    }
  end
end

File: sdk/csharp/src/Dagger.SDK.CodeGen/Types/EnumValue.cs
using System.Text.Json.Serialization;

namespace Dagger.SDK.CodeGen.Types;

public class EnumValue
{
    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("description")]
    public required string Description { get; set; }

    [JsonPropertyName("isDeprecated")]
    public bool IsDeprecated { get; set; }

    [JsonPropertyName("directives")]
    public Directive[]? Directives { get; set; }
}
